js/node_shared/Semgrep_node_js_shared.ml

```diff
@@ -0,0 +1,85 @@
+open Js_of_ocaml
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+
+(* What is this folder for? We may have some common functionalities that we
+   would like to share among purely the Node-specific Javascript code. For
+   instance, our `unix.js` is something which should not be included into all
+   of the JSCaml code (like the parsers), but just the test code and actual
+   LSP code.
+
+   To maek this distinction apparent, we have `shared`, which is for things
+   universal to all of the JSCaml code, and `node_shared`, for only things
+   which should be distributed via Node.js.
+*)
+
+(*****************************************************************************)
+(* Reporting *)
+(*****************************************************************************)
+
+(* JS specific IO for the RPC server *)
+module Io = RPC_server.MakeLSIO (struct
+  type input = in_channel
+  type output = out_channel
+
+  let read_line ic = input_line ic |> Lwt.return_some
+  let stdin = stdin
+  let stdout = stdout
+  let flush () = flush stdout |> Lwt.return
+  let atomic f oc = f oc
+
+  let write _ str =
+    (* nosem *)
+    print_string str;
```

Comment:
why `print_string` instead of explicitly writing to the `stdout` channel?

Reply:
(might be more of an @ajbt200128 question since I realize now the code was moved from `.../language_server/Main.ml`)